benchmarkapps/PlatformBenchmarks/BenchmarkApplication.HttpConnection.cs

```diff
@@ -72,39 +72,39 @@ private async Task ProcessRequestsAsync()
                     }
 
                     // No more input or incomplete data, Advance the Reader
-                    Reader.AdvanceTo(buffer.Start, examined);
+                    Reader.AdvanceTo(consumed, examined);
                     break;
                 }
             }
         }
 
-        private bool ParseHttpRequest(ref ReadOnlySequence<byte> buffer, bool isCompleted, out SequencePosition examined)
+        private bool ParseHttpRequest(ref ReadOnlySequence<byte> buffer, bool isCompleted, out SequencePosition consumed, out SequencePosition examined)
         {
             examined = buffer.End;
-
-            var consumed = buffer.Start;
+            consumed = buffer.Start;
             var state = _state;
+            var reader = new BufferReader<byte>(buffer);
 
-            if (!buffer.IsEmpty)
+            if (!reader.End)
             {
                 if (state == State.StartLine)
                 {
-                    if (Parser.ParseRequestLine(new ParsingAdapter(this), buffer, out consumed, out examined))
+                    if (Parser.ParseRequestLine(new ParsingAdapter(this), ref reader))
                     {
                         state = State.Headers;
+                        consumed = reader.Position;
+                        examined = consumed;
                     }
-
-                    buffer = buffer.Slice(consumed);
                 }
 
                 if (state == State.Headers)
                 {
-                    if (Parser.ParseHeaders(new ParsingAdapter(this), buffer, out consumed, out examined, out int consumedBytes))
+                    if (Parser.ParseHeaders(new ParsingAdapter(this), ref reader))
                     {
                         state = State.Body;
+                        consumed = reader.Position;
+                        examined = consumed;
                     }
-
-                    buffer = buffer.Slice(consumed);
```

Comment:
Removing these two slices of the ref buffer param always results in an infinite loop in BenchmarkApplication.ProcessRequestsAsync() since buffer.IsEmpty is never becomes true.

Reply:
Add test to catch this? (e.g. count iterations)

Reply:
> Add test to catch this? (e.g. count iterations)

Not sure where that would go.

Reply:
> Not sure where that would go.

[HttpParserTests.cs](https://github.com/aspnet/KestrelHttpServer/blob/master/test/Kestrel.Core.Tests/HttpParserTests.cs)?

Something based on [ProcessRequestsAsync](https://github.com/aspnet/KestrelHttpServer/blob/83488886e0ed32e946af2a25ae7f29ce9bcc4273/benchmarkapps/PlatformBenchmarks/BenchmarkApplication.HttpConnection.cs#L40-L79) ensuring its not stuck in an infinite loop?

Reply:
I don't see that we have a reference to it anywhere from tests. I'm a little hesitant to start adding them given my lack of experience in this codebase...